### PR TITLE
fix: handle providers with no available models to prevent crashes

### LINKED ISSUE #5858: TypeError: undefined is not an object crash on start after v1.0.180 update
### Description

After the latest update to v1.0.180, I get the following crash on startup:

```
{
  "name": "UnknownError",
  "data": {
    "message": "TypeError: undefined is not an object (evaluating 'Provider2.sort(Object.values(item.models))[0].id')\n    at <anonymous> (src/server/server.ts:1604:95)\n    at o3 (../../node_modules/.bun/remeda@2.26.0/node_modules/remeda/dist/chunk-3ZJAREUD.js:1:137)\n    at <anonymous> (src/server/server.ts:1604:22)\n    at processTicksAndRejections (native:7:39)"
  }
}
```

I have tried to downgrade to different versions of opencode and now only get this error no matter the version.

### OpenCode version

v1.0.180

### Steps to reproduce

1. Install opencode v1.0.180
2. Run `opencode` to start a new session
3. Get this crash

### Screenshot and/or share link

_No response_

### Operating System

macOS 15.7.1

### Terminal

Ghostty w/ zsh

### PR DESCRIPTION
**Pull request by ahosker**
## Summary
- Fix runtime crashes when providers have no available models
- Add validation in provider merge process
- Prevent empty model arrays from breaking ACP agent functionality
- Graceful degradation with error logging for debugging

## Fixes
- Prevents crashes in model selection dialogs
- Handles edge cases in provider configuration merges
- Improves error handling in server API endpoints

## Test
OpenCode continues to work against the earlier failed `models.json` (attatched)
[models_broken.json](https://github.com/user-attachments/files/24274718/models_broken.json)

Closes #5858

### CODE REVIEW

File: packages/opencode/src/server/server.ts [hunk 1]
```diff
@@ -1481,7 +1481,14 @@ export namespace Server {
...
             providers: Object.values(providers),
-            default: mapValues(providers, (item) => Provider.sort(Object.values(item.models))[0].id),
+            default: mapValues(providers, (item) => {
+              const sorted = Provider.sort(Object.values(item.models))
+              if (sorted.length === 0) {
+                log.error("Provider has no models available", { providerID: item.id })
+                return null
+              }
+              return sorted[0].id
+            }),
```
[the remark this answers is not in the document]
Comment (ahosker) [reply packages/opencode/src/server/server.ts]: type mismatch

**Comment by ahosker** [comment]
Committed the type safety fixes.

File: packages/opencode/src/server/server.ts [hunk 2]
```diff
@@ -1481,7 +1481,18 @@ export namespace Server {
...
+            default: Object.fromEntries(
+              Object.entries(providers)
+                .map(([providerID, item]) => {
+                  const sorted = Provider.sort(Object.values(item.models))
+                  if (sorted.length === 0) {
+                    log.error("Provider has no models available", { providerID })
+                    return null
+                  }
+                  return [providerID, sorted[0].id]
+                })
```
[the remark this answers is not in the document]
Comment (ahosker) [reply packages/opencode/src/server/server.ts:3407]: Can do, however the style guide says.

Functions: Keep logic within a single function unless breaking it out adds clear reuse or composition benefits.

**Comment by ahosker** [comment]
Too told, too many changes since this was written.
